File: src/Projects/Android/ChatSystem/ChatSystemRepositoryLib/src/main/java/org/csystem/app/chatsystem/repository/entity/LoginInfo.java
package org.csystem.app.chatsystem.repository.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "login_info")
public class LoginInfo {
    @Id
    @Column(name = "login_info_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long id;

    @Column(name = "date_time", nullable = false)
    public LocalDateTime localDateTime = LocalDateTime.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "nick_name", nullable = false)
    public User user;

    @Override
    public int hashCode()
    {
        return Long.hashCode(id);
    }

    @Override
    public boolean equals(Object other)
    {
        return other instanceof LoginInfo li && li.id == id;
    }
}

File: src/Projects/Android/PaymentApplication/RepositoryLib/src/main/java/org/csystem/android/app/payment/repository/entity/LoginInfo.kt
package org.csystem.android.app.payment.repository.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.time.LocalDateTime

@Entity(tableName = "login_info")
data class LoginInfo(@PrimaryKey var id: Long = 0,
                     var username: String = "",
                     var password: String = "",
                     var success: Boolean = true,
                     @ColumnInfo(name = "login_date_time") var loginDateTime: LocalDateTime = LocalDateTime.now()) : Serializable